tests/common/test_parser.py

```diff
@@ -107,3 +108,6 @@ def test_dataclass_from_dict():
     # test extra keys not silently ignored in strict mode
     with pytest.raises(KeyError):
         dataclass_from_dict(BasicModel, raw2, strict=True)
+    # test int gets converted to bytes
+    res = parse_value("int_value_that_should_be_bytes", 5, bytes)
+    assert isinstance(res, bytes)
```

Comment:
Maybe assert what the value is instead?

Reply:
Yeah and I just realize this conversion is wrong
bytes(5) creates b"\x00\x00\x00\x00\x00" and not b'\x05\x00\x00\x00'

Reply:
https://docs.python.org/3/library/stdtypes.html#int.to_bytes

Reply:
Yeah exactly but still I wonder if we should do it this way. I just looked up the spec and this should contain the NOC as (bytes encoded) string. So a single int in this field is basically just junk and we should just ignore it. We can parse it to bytes (assuming little endian, unsigned) but its still junk.

Reply:
anyways, changed it so we're unblocked.

Reply:
@MartinHjelmare what do you think ? Do the conversion (as at least we get a bytes object back) or just set it to an empty bytes object ?

Reply:
What does the NOC do?

Reply:
> What does the NOC do?

Its a certificate

Reply:
So setting it to an empty string feels more safe here to me

Reply:
I changed it to only handle this specific edge case - it didn't feel right otherwise as we would be hiding issues.